pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentStatsHistory.java

```diff
@@ -253,23 +267,27 @@ public String toString() {
     return "cursor=" + getCursor() + ",numEntries=" + getArraySize() + ",isFull=" + isFull();
   }
 
-  public synchronized void serializeInto(File outFile) throws IOException {
-    try (OutputStream os = new FileOutputStream(outFile);
+  public synchronized void save() {
+    try (OutputStream os = new FileOutputStream(new File(_historyFilePath));
```

Comment:
Is this using java serialization or is writeObject overridden elsewhere? If former, may be better to avoid that.